Automattic/wp-calypso
# Site Migration: Add `from` param to Tracks props on Upgrade screen

Related to #91019

## Proposed Changes

* Add `from` and `has_source_site` props to the Tracks event properties for the `calypso_site_migration_upgrade_plan_screen` event so we can track how many people get to this point in the flow without a `from`/source site.
* It might be overkill to have both of these? Can we rely just on the `from` property?

## Why are these changes being made?


* https://github.com/Automattic/wp-calypso/issues/91019

## Testing Instructions



* Switch to this PR
* Navigate to `/start` and make sure you're showing Tracks events in your browser console (or using Tracks Vigilante)
* Choose Import at the goals step
* Put in your URL
* Choose Everything
* Check the upgrade plan step Tracks event in your browser; it should have a `from` property with the site you entered, and the `has_source_site` property should be `true`:

<img width="962" alt="Screenshot 2024-05-23 at 4 55 58 PM" src="https://github.com/Automattic/wp-calypso/assets/2124984/bf359c96-421d-464f-83c2-937fd3b86a09">

* Now go through the flow from `/start` again
* Select Import at the goals step
* Do not put a URL in at the "Let’s import your content" step; choose "Pick your platform from a list" instead
* Choose WordPress
* Choose Everything
* On the upgrade plan step, the Tracks event should not have a value in the `from` property, and `has_source_site` should be `false`:

<img width="875" alt="Screenshot 2024-05-23 at 4 58 01 PM" src="https://github.com/Automattic/wp-calypso/assets/2124984/2f00dcaf-87fc-4527-94d6-6ed3c1ba1750">



## Pre-merge Checklist
